### DISCUSSION TITLE
kalkancrypl и OCSP валидация.

### DISCUSSION THREAD

**Discussion by TrunovK** [malikzh/NCANode#165] [Q&A]
Добрый день.

Выполняю проверку cms файлов и пользовательских ключей через метод OCSP и получаю следующую ошибку:
```
{
"revoked": false,
"by": "OCSP",
"revocationTime": null,
"reason": "malformed response: unknown object in factory: kz.gov.pki.kalkan.asn1.DERUnknownTag"
}
```

Предварительно файлы kalkancrypt_xmldsig-0.4.jar и kalkancrypt-0.7.2.jar положены в папку lib, билд на 17 JDK успешно прошел.
Образ докера успешно запущен командами
```
docker volume create ncanode_cache
docker run -p 14579:14579 -v ncanode_cache:/app/cache -d malikzh/ncanode
```

Валидация CRL проходит успешно, а OCSP нет. В чем может быть проблема?

**Comment by malikzh** [comment]
Суть проблемы: Почему-то библиотека kalkan не может распознать ответ от сервера. А почему это происходит я не знаю, это надо смотреть ответ. 

Попробуйте в отладчик в IDEA например поставить breakpoint здесь:

https://github.com/malikzh/NCANode/blob/master/src/main/java/kz/ncanode/service/OcspService.java#L92

И сохраните значения входящих аргументов метода. Вот эти переменные:

``` java
private OcspStatus processOcspResponse(InputStream response, byte[] nonce) // <- эти два
```

И прикрепите здесь попробуем разобраться. Потому что я не смогу повторить эту ошибку у себя.


З.Ы. Преобразовать `InputStream` в `byte[]` можно так:

```java
byte[] ocspResultBytes = response.readAllBytes();
```

**Reply by TrunovK** [reply]
OCSP response: 
[48, -126, 6, 56, 10, 1, 0, -96, -126, 6, 49, 48, -126, 6, 45, 6, 9, 
43, 6, 1, 5, 5, 7, 48, 1, 1, 4, -126, 6, 30, 48, -126, 6, 26, 48, -126, 1, -47, -95, -126, 1, 11, 48, -126, 1, 7, 49, 23, 48, 21, 6, 3, 85, 4, 3, 12, 14, 79, 67, 83, 80, 32, 82, 69, 83, 80, 79, 78, 68, 69, 82, 49, 24, 48, 22, 6, 3, 85, 4, 5, 19, 15, 73, 73, 78, 55, 54, 49, 50, 51, 49, 51, 48, 48, 51, 49, 51, 49, 11, 48, 9, 6, 3, 85, 4, 6, 19, 2, 75, 90, 49, 21, 48, 19, 6, 3, 85, 4, 7, 12, 12, -48, -112, -48, -95, -48, -94, -48, -112, -48, -99, -48, -112, 49, 21, 48, 19, 6, 3, 85, 4, 8, 12, 12, -48, -112, -48, -95, -48, -94, -48, -112, -48, -99, -48, -112, 49, 24, 48, 22, 6, 3, 85, 4, 11, 12, 15, 66, 73, 78, 48, 48, 48, 55, 52, 48, 48, 48, 48, 55, 50, 56, 49, 125, 48, 123, 6, 3, 85, 4, 10, 12, 116, -48, -112, -48, -102, -48, -90, -48, -104, -48, -98, -48, -99, -48, -107, -48, -96, -48, -99, -48, -98, -48, -107, 32, -48, -98, -48, -111, -48, -87, -48, -107, -48, -95, -48, -94, -48, -110, -48, -98, 32, 34, -48, -99, -48, -112, -48, -90, -48, -104, -48, -98, -48, -99, -48, -112, -48, -101, -48, -84, -48, -99, -48, -85, -48, -107, 32, -48, -104, -48, -99, -48, -92, -48, -98, -48, -96, -48, -100, -48, -112, -48, -90, -48, -104, -48, -98, -48, -99, -48, -99, -48, -85, -48, -107, 32, -48, -94, -48, -107, -48, -91, -48, -99, -48, -98, -48, -101, -48, -98, -48, -109, -48, -104, -48, -104, 34, 24, 15, 50, 48, 50, 51, 48, 52, 49, 51, 48, 57, 52, 56, 50, 51, 90, 48, -127, -128, 48, 126, 48, 105, 48, 13, 6, 9, 96, -122, 72, 1, 101, 3, 4, 2, 1, 5, 0, 4, 32, 4, 25, 8, 84, 87, -3, -62, 21, 57, -3, -60, 28, -40, 120, 86, -38, 14, -76, -3, 33, -104, -120, 74, 59, 34, -61, -6, -27, 84, -93, 100, 13, 4, 32, -37, 120, 14, 11, 48, 97, 38, -31, -90, 115, -27, 45, -83, 59, -127, 121, 11, 125, 20, 64, -11, 66, -121, 30, -85, 85, -96, -84, 54, 13, 72, -50, 2, 20, 72, 105, 123, -109, 122, 10, 90, -68, -115, 124, -50, -56, 69, -107, -58, 110, -14, -119, 59, -121, -128, 0, 24, 15, 50, 48, 50, 51, 48, 52, 49, 51, 48, 57, 52, 56, 50, 51, 90, -95, 44, 48, 42, 48, 23, 6, 9, 43, 6, 1, 5, 5, 7, 48, 1, 2, 4, 10, 4, 8, -77, -11, -54, 66, 103, 21, 0, -5, 48, 15, 6, 9, 43, 6, 1, 5, 5, 7, 48, 1, 9, 4, 2, 5, 0, 48, 13, 6, 9, 42, -125, 14, 3, 10, 1, 1, 1, 2, 5, 0, 3, 65, 0, 114, -39, 105, -3, -6, 87, -14, -88, -45, 70, 4, 104, -89, 44, -92, -30, -91, 86, -118, -70, -69, -91, 90, -122, 2, 23, 72, -100, -80, -119, -40, 94, 117, -80, -3, 31, 78, -57, 71, -9, -59, -27, 58, -50, -114, 60, -41, -78, -43, 89, 58, -123, 64, 73, -117, -75, -44, -52, 30, -125, -64, 63, -24, 37, -96, -126, 3, -17, 48, -126, 3, -21, 48, -126, 3, -25, 48, -126, 3, -111, -96, 3, 2, 1, 2, 2, 20, 115, -107, 43, 26, 21, -10, 40, 54, -90, 68, -92, -61, -29, -53, 101, -48, 39, -15, 60, -33, 48, 13, 6, 9, 42, -125, 14, 3, 10, 1, 1, 1, 2, 5, 0, 48, 83, 49, 11, 48, 9, 6, 3, 85, 4, 6, 19, 2, 75, 90, 49, 68, 48, 66, 6, 3, 85, 4, 3, 12, 59, -46, -80, -48, -101, -48, -94, -48, -94, -48, -85, -46, -102, 32, -48, -102, -48, -93, -45, -104, -48, -101, -48, -112, -48, -99, -48, -108, -48, -85, -48, -96, -48, -93, -48, -88, -48, -85, 32, -48, -98, -48, -96, -48, -94, -48, -112, -48, -101, -48, -85, -46, -102, 32, 40, 71, 79, 83, 84, 41, 48, 30, 23, 13, 50, 50, 49, 48, 49, 56, 48, 52, 50, 57, 50, 52, 90, 23, 13, 50, 51, 49, 48, 49, 56, 48, 52, 50, 57, 50, 52, 90, 48, -126, 1, 7, 49, 23, 48, 21, 6, 3, 85, 4, 3, 12, 14, 79, 67, 83, 80, 32, 82, 69, 83, 80, 79, 78, 68, 69, 82, 49, 24, 48, 22, 6, 3, 85, 4, 5, 19, 15, 73, 73, 78, 55, 54, 49, 50, 51, 49, 51, 48, 48, 51, 49, 51, 49, 11, 48, 9, 6, 3, 85, 4, 6, 19, 2, 75, 90, 49, 21, 48, 19, 6, 3, 85, 4, 7, 12, 12, -48, -112, -48, -95, -48, -94, -48, -112, -48, -99, -48, -112, 49, 21, 48, 19, 6, 3, 85, 4, 8, 12, 12, -48, -112, -48, -95, -48, -94, -48, -112, -48, -99, -48, -112, 49, 
24, 48, 22, 6, 3, 85, 4, 11, 12, 15, 66, 73, 78, 48, 48, 48, 55, 52, 48, 48, 48, 48, 55, 50, 56, 49, 125, 48, 123, 6, 3, 85, 4, 10, 12, 116, -48, -112, -48, -102, -48, -90, -48, -104, -, -48, -90, -48, -104, -48, -98, -48, -99, -48, -112, -48, -101, -48, -84, -48, -99, -48, -85, -48, -107, 32, -48, -104, -48, -99, -48, -92, -48, -98, -48, -96, -48, -100, -48, -112, -48, -90, -48, -104, -48, -98, -48, -99, -48, -99, -48, -85, -48, -107, 32, -48, -94, -48, -107, -48, -91, -48, -99, -48, -98, -48, -101, -48, -98, -48, -109, -48, -104, -48, -104, 34, 48, 108, 48, 37, 6, 9, 42, -125, 14, 3, 10, 1, 1, 1, 1, 48, 24, 6, 10, 42, -125, 14, 3, 10, 1, 1, 1, 1, 1, 6, 10, 42, -125, 14, 3, 10, 1, 3, 1, 1, 0, 3, 67, 0, 4, 64, -24, -42, -63, -67, 
64, 58, 85, 30, 115, 19, 82, -59, 31, 42, -15, 98, 98, -3, 84, -49, 61, 116, -52, -14, -125, -58, -70, 28, -42, 119, -39, 13, -46, -3, -33, 43, -120, -36, 22, 35, -107, -78, 114, 80, 103, -50, -1, -19, 51, -55, 19, 106, 50, -47, -117, -117, -74, 115, -109, 44, 79, -22, 77, -104, -93, -126, 1, 117, 48, -126, 1, 113, 48, 92, 6, 3, 85, 29, 46, 4, 85, 48, 83, 48, 81, -96, 79, -96, 77, -122, 36, 104, 116, 116, 112, 58, 47, 47, 99, 114, 108, 46, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 47, 110, 99, 97, 95, 100, 95, 103, 111, 115, 116, 46, 99, 114, 
108, -122, 37, 104, 116, 116, 112, 58, 47, 47, 99, 114, 108, 49, 46, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 47, 110, 99, 97, 95, 100, 95, 103, 111, 115, 116, 46, 99, 114, 108, 
48, 99, 6, 8, 43, 6, 1, 5, 5, 7, 1, 1, 4, 87, 48, 85, 48, 47, 6, 8, 43, 6, 1, 5, 5, 7, 48, 2, -122, 35, 104, 116, 116, 112, 58, 47, 47, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 47, 99, 101, 114, 116, 47, 110, 99, 97, 95, 103, 111, 115, 116, 46, 99, 101, 114, 48, 34, 6, 8, 43, 6, 1, 5, 5, 7, 48, 1, -122, 22, 104, 116, 116, 112, 58, 47, 47, 111, 99, 115, 112, 46, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 48, 13, 6, 9, 43, 6, 1, 5, 5, 7, 48, 1, 5, 4, 0, 48, 19, 6, 3, 85, 29, 37, 4, 12, 48, 10, 6, 8, 43, 6, 1, 5, 5, 7, 3, 9, 48, 15, 6, 3, 85, 29, 35, 4, 8, 48, 6, -128, 4, 91, 106, 115, -23, 48, 29, 6, 3, 85, 29, 14, 4, 22, 4, 20, -105, -105, 127, -123, -100, 105, -93, -93, 117, -73, 14, 70, -70, 55, 103, 45, -50, -27, -17, 94, 48, 88, 6, 3, 85, 29, 31, 4, 81, 48, 79, 48, 77, -96, 75, -96, 73, -122, 34, 104, 116, 116, 112, 58, 47, 47, 99, 114, 108, 46, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 47, 
110, 99, 97, 95, 103, 111, 115, 116, 46, 99, 114, 108, -122, 35, 104, 116, 116, 112, 58, 47, 47, 99, 114, 108, 49, 46, 112, 107, 105, 46, 103, 111, 118, 46, 107, 122, 47, 110, 99, 97, 95, 103, 111, 115, 116, 46, 99, 114, 108, 48, 13, 6, 9, 42, -125, 14, 3, 10, 1, 1, 1, 2, 5, 0, 3, 65, 0, 126, -9, -107, 70, -4, -99, -33, -95, 103, 36, 57, -124, -49, 91, 15, 114, -128, 
87, -36, -87, 23, 2, 88, -2, -112, -27, -17, 94, 30, -59, 73, -62, -51, 15, -24, 45, -122, -121, 11, 59, -39, 94, -37, 82, 104, 11, -94, 91, -109, 67, -63, 8, -107, 64, -98, -118, -92, 
-21, 110, 82, 26, 2, 31, -30], 

nonce [-77, -11, -54, 66, 103, 21, 0, -5]

**Comment by TrunovK** [comment]
Добрый день. Есть небольшие новости. Развернул собранный образ на рабочем ренчере в продуктивной среде. И там все заработало. Локально на моем компьютере ошибка еще осталась
